src/main/java/app/linguistai/bmvp/service/gamification/FriendshipService.java

```diff
@@ -0,0 +1,178 @@
+package app.linguistai.bmvp.service.gamification;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import app.linguistai.bmvp.model.Friendship;
+import app.linguistai.bmvp.repository.gamification.IFriendshipRepository;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.enums.FriendshipStatus;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Service
+public class FriendshipService {
+
+    private final IAccountRepository accountRepository;
+    private final IFriendshipRepository friendshipRepository;
+
+    @Transactional
+    public Friendship sendFriendRequest(String user1Email, UUID user2Id) throws Exception {
+        try {
+            User dbUser1 = accountRepository.findUserByEmail(user1Email).orElseThrow(() -> new Exception("User is not found"));
+            User dbUser2 = accountRepository.findUserById(user2Id).orElseThrow(() -> new Exception("Requested user is not found"));
+
+            if (dbUser1.getId().equals(dbUser2.getId())) {
+                throw new Exception("User cannot send friend request to themselves");
+            }
+
+            // check if friend request exists
+            Friendship friendship = friendshipRepository.findByUserPair(dbUser1.getId(), user2Id).orElse(null);
+
+            if (friendship != null) {
+                throw new Exception("Friendship or request already exist");
+            }
+            
+            LocalDateTime now = LocalDateTime.now();
+
+            // save friendship to the db with pending status
+            friendship = friendshipRepository.save(new Friendship(dbUser1, dbUser2, now, FriendshipStatus.PENDING));
+
+            return friendship;
+        } catch (Exception e) {
+            System.out.println(e.getLocalizedMessage());
+            throw e;
+        }
+    }
+
+    public List<Friendship> getFriends(String userEmail) throws Exception {
+        try {
+            User dbUser1 = accountRepository.findUserByEmail(userEmail).orElse(null);
+
+            if (dbUser1 == null) {
+                throw new Exception("User is not found");
+            }
+
+            List<Friendship> friends = friendshipRepository.findByUser1OrUser2AndStatus(dbUser1.getId(), FriendshipStatus.ACCEPTED);
+
+            return friends;
+        } catch (Exception e) {
+            System.out.println(e.getLocalizedMessage());
+            throw e;
+        }
+    }
+
+    public List<Friendship> getFriendRequests(String userEmail) throws Exception {
+        try {
+            User dbUser1 = accountRepository.findUserByEmail(userEmail).orElse(null);
+
+            if (dbUser1 == null) {
+                throw new Exception("User is not found");
```

Comment:
Just as a comment, this can be replaced by replacing `orElse()` above with `orElseThrow()` for a more readable code, but it is also fine to leave like this